lib/adobexd.js

```diff
@@ -43,22 +43,42 @@ function filterReferences(references, filter) {
   return { key: filter, elements: filteredElements };
 }
 
+function findArtboardByID(artboards, id) {
+  const board = artboards.find(b => b.id === id.replace('artboard-', '')); 
+
+  return board.artboard.children[0];
+}
+
 async function parseXDFile(xdFilePath, outputDir) {
-  
   await unzipFile(xdFilePath, outputDir);
 
   const MANIFEST_FILEPATH = path.join(outputDir, 'manifest');
   const xdManifest = JSON.parse(await readFile(MANIFEST_FILEPATH));
 
-  const artboards = await getArtboardFiles(outputDir);
-
   const tokenArtboards = ['primitives/color', 'primitives/blur', 'primitives/border', 'primitives/gradient', 'textstyle'];
   const tokenReferences = getReferencesFromManifest(xdManifest);
   const groupedTokenReferences = tokenArtboards.map(key => filterReferences(tokenReferences, key));
 
+  const artboards = await getArtboardFiles(outputDir);
+
+  const matchedTokens = groupedTokenReferences.map(tokenGroup => tokenGroup.elements.map(ref => {
+    const { name } = ref;
+    const value = findArtboardByID(artboards, ref.pathToToken); 
+
+    return { name, value };
+  }));
+
+  let [colors, blurs, borders, gradients, textStyles] = matchedTokens;
```

Comment:
hmm. are they always going to be in this order? If the order changes now, you got a big issue.

Reply:
It's the same order as the array in line 58